commons/src/scanner/mod.rs

```diff
@@ -168,64 +223,59 @@ impl From<&str> for SymbolKind {
 }
 
 /// Trait for conversion of input into Unimarkup symbols.
-pub trait IntoSymbols<'s, T> {
-    /// Converts input into Unimarkup symbols.
-    fn into_symbols(self) -> T;
+pub trait IntoSymbols<'s> {
+    type Output;
+
+    /// Converts input into collection of Unimarkup symbols.
```

Comment:
You're right, this can return anything basically. The only restriction is at the use site, and that can guide users of the API to write the "right" implementation. 

However, I'm thinking about reworking this API completely. Aside from the problem you mentioned, here's also the problem that we have to pass it config at a later point (because of localisation). The way it's used right now won't make that possible. 

Possible solution would be to create a `Scanner` struct, with some trait (for example `trait Scan`) that creates a Scanner from input. We could even provide blanket implementation of Scanner for `T: AsRef<str>`. We can then provide functions to create create iterator from scanner, provide some collections like slice etc. That way the `Output` can be completely controlled by us, and we can have a default Config for Scanner, or pass it specific Config if needed.

What do you think?

Reply:
Could work. I guess this all depends on how we are able to handle localisation for the grapheme segmenter.

Reply:
I will try it in a separate PR, just so we can clearly see if, how and why does it improve the API.